### PR TITLE
feat: add ledoit-wolf-shrinkage task (medium)

### PR DESCRIPTION
**Pull request by yyu56253**
**If your PR is adding a new task to QuantitativeFinance-Bench, please complete this by adding an "x" next to each applicable item.**
- [x] I ran `harbor check tasks/ledoit-wolf-shrinkage` on my new task and ensured all checks pass
- [x] My `instruction.md` was written by a human
- [x] All behavior checked in tests is described in `instruction.md`
- [x] All behavior described in `instruction.md` is checked in tests
- [x] `instruction.md` does NOT mention which skills to use (agents must discover skills themselves)
- [x] My test cases have informative docstrings that describe which behavior they check
- [x] It is hard for the agent to cheat (e.g., editing data files, looking inside files for solution strings, etc.)
- [x] My `task.toml` was written by a human
- [x] My `solution/solve.sh` was written by a human (with minimal help from a language model)
- [x] If external dependencies are used, versions are pinned for reproducibility
- [x] If the agent produces structured data (API, JSON, CSV, etc.), the exact schema is documented in instruction.md or a referenced spec file
- [x] Model Validation: I have verified this task using at least one model from the Frontier Tier (Claude Sonnet 4.5) in the `Agent Runs` section below
- [x] I have provided a root cause analysis for every unit test failure for each agent run in the `Agent Runs` section below (no unit tests failed)

**This task does not include skills** — skills-related checkboxes are N/A.

---

## Summary

Adds a medium-difficulty covariance-estimation task. The agent computes the Ledoit-Wolf (2004) constant-correlation shrinkage estimator from 2 years of daily returns for 10 assets, then builds unconstrained minimum-variance portfolios from both the shrunk and sample covariance matrices and compares annualized volatilities.

**Skills demonstrated**: NaN-row dropping, `1/T` (not `1/(T-1)`) sample covariance, constant-correlation shrinkage target construction, Ledoit-Wolf optimal shrinkage intensity (pi/rho/gamma computation with asymptotic variance/covariance terms), matrix inversion for unconstrained min-var weights, and annualized portfolio volatility.

## Task Metadata

| Field | Value |
|-------|-------|
| **Task ID** | `ledoit-wolf-shrinkage` |
| **Difficulty** | Medium — multi-step estimation pipeline (clean → sample cov → target → shrinkage intensity → shrunk cov → min-var portfolio). The pi/rho/gamma terms in the LW2004 formula have multiple interacting loops that are easy to miscalculate. |
| **Category** | `cross-domain` (statistical estimation + portfolio optimization) |
| **Skills Provided** | None |
| **Data Source of Input Files** | Synthetic multivariate normal returns from a block-correlated structure (seed-fixed, deterministic), with 3 NaN values injected. |

## Agent Performance

| Agent | Model | Trials | Tests Passed | Reward |
|-------|-------|--------|--------------|--------|
| oracle       | —                      | 1 | 33/33 | 1.000 |
| claude-code  | claude-sonnet-4-5      | 1 | 33/33 | **1.000** |
| Finance-Zero | gemini/gemini-2.5-flash-lite | 3 (`-k 3`) | — | 0.000 |

### Harbor tasks check
All 11/11 checks pass (numpy==1.26.4 and pandas==2.2.3 pinned; pytest in if-condition for reward.txt).

### Oracle Run
```
$ harbor run --path ./tasks --include-task-name ledoit-wolf-shrinkage --agent oracle
  1/1 Mean: 1.000   33 passed   reward = 1.0
```

### Agent Runs — Claude Sonnet 4.5
```
$ harbor run --path ./tasks --include-task-name ledoit-wolf-shrinkage \
    --agent claude-code --model anthropic/claude-sonnet-4-5-20250929 \
    --ae ANTHROPIC_API_KEY=...
  1/1 Mean: 1.000   Trials: 1, Errors: 0, reward = 1.0
```
All 33 unit tests passed. No failures to root-cause-analyze.

### Finance-Zero — Error Analysis

| Model | Trials | Reward |
|-------|--------|--------|
| gemini/gemini-2.5-flash-lite | 3 (`-k 3`) | 0.000 × 3 |

**Root cause**: The generated script runs to completion and produces `results.json`, but the numerical values are incorrect. The Ledoit-Wolf (2004) shrinkage intensity formula requires computing three interacting quantities (pi, rho, gamma) with nested summation loops including cross-product terms (`theta_ij`, `theta_ji`) and sqrt-ratio scaling factors. The agent's one-shot implementation gets the `rho` computation wrong, producing an incorrect shrinkage intensity `delta*` that cascades into the shrunk covariance matrix and all downstream portfolio weights.

**Per-test outcome**: 0/33 tests pass (reward = 0.0).

## Notes

- The `1/T` covariance normalization (not `1/(T-1)`) is load-bearing for the shrinkage formula. Using ddof=1 shifts the shrinkage intensity by ~5% and breaks multiple weight tests.
- The instruction explicitly prohibits `sklearn.covariance.LedoitWolf` — the agent must implement the estimator from scratch.
- No skills included by design.

### CODE REVIEW

**Review by Dongzhikang — COMMENTED** [review]
## Review: feat: add ledoit-wolf-shrinkage task (medium)
Reviewer: Grim 🔍 | Date: 2026-04-15

I pulled the PR locally and reviewed `instruction.md`, `solution/solve.sh`, and `tests/test_outputs.py`.

I did not find a clear blocker in the current version. The task definition, oracle, and tests appear broadly aligned on:
- dropping NaN rows,
- using the `1/T` covariance normalization,
- constructing the constant-correlation target,
- computing the Ledoit-Wolf shrinkage intensity from the stated formula,
- and producing the sample vs. shrunk minimum-variance portfolios.

A couple of things I checked specifically:
- the data really does have 3 NaN-containing rows, so the `n_observations = 501` expectation is consistent with the task data;
- the trace-preservation test is appropriate here because the shrinkage target preserves the sample diagonal by construction.

### Verdict
APPROVE

I did not find a correctness issue that should block merge.

**Comment by XiangningLin** [comment]
LGTM

**Review by XiangningLin — APPROVED** [review]
